### DISCUSSION TITLE
[request] Implement an interface class so we can implement our own DebugRenderer

### DISCUSSION THREAD

**Discussion by RevoluPowered** [jrouwe/JoltPhysics#725] [Q&A] [answered]
I would not mind if there was some dumb interface like IDebugRenderer we could implement in our game engine?

We could then implement our own Surface renderer directly and render using our native rendering solution.

Someone could come along and make a platform universal version in godot/sdl/etc

**Comment by mihe** [comment, marked as answer]
This already exists:

https://github.com/jrouwe/JoltPhysics/blob/9d63f5a2d1b9e426a54dce6c8979e22ff6004eba/Jolt/Renderer/DebugRenderer.h#L28-L29

You implement the virtual functions there and then call any of these methods on the `PhysicsSystem`:

https://github.com/jrouwe/JoltPhysics/blob/9d63f5a2d1b9e426a54dce6c8979e22ff6004eba/Jolt/Physics/PhysicsSystem.h#L127-L137

As a reference you have the implementation used by the `Samples` application here:

https://github.com/jrouwe/JoltPhysics/blob/9d63f5a2d1b9e426a54dce6c8979e22ff6004eba/TestFramework/Renderer/DebugRendererImp.h#L27-L28

**Comment by RevoluPowered** [comment]
Thank you, this is great. I'll implement this next week. :)
